src/back/TicketManager/src/Core/TicketManager.Core.Services/Endpoints/Events/EventHoldEndpoint.cs

```diff
@@ -0,0 +1,39 @@
+using FastEndpoints;
+using TicketManager.Core.Contracts.Events;
+using TicketManager.Core.Domain.Accounts;
+using TicketManager.Core.Domain.Events;
+using TicketManager.Core.Services.DataAccess.Repositories;
+
+namespace TicketManager.Core.Services.Endpoints.Events;
+
+public class EventHoldEndpoint: Endpoint<EventStatusManipulationRequest>
+{
+    private readonly Repository<Event, Guid> events;
+
+    public EventHoldEndpoint(Repository<Event, Guid> events)
+    {
+        this.events = events;
+    }
+
+    public override void Configure()
+    {
+        Post("/event/hold");
+        Roles(AccountRoles.Organizer, AccountRoles.Admin);
+    }
+
+    public override async Task HandleAsync(EventStatusManipulationRequest req, CancellationToken ct)
+    {
+        var @event = await events.FindAndEnsureExistenceAsync(req.Id, ct);
+        try
+        {
+            @event.ChangeEventStatus(EventStatus.Held);
+        }
+        catch (Exception)
+        {
+            await SendErrorsAsync(cancellation: ct);
+            return;
+        }
```

Comment:
If you create validator, you don't have to catch exception in endpoint handler. You have to create validator anyway to check if event even exists.

Reply:
This try catch is not to check whether `@event` exists. It has to do with `ChangeEventStatus` logic which throws exceptions when invalid status change is requested. Do you want me to once again check this logic in validator?

Reply:
Yea. I don't like mixing validation and logic. Sometimes in this project we have to do that, because of our api specification, but generally I don't like it. Also in every single online tutorial people extract validation to separate class to not extend handler too much.

Reply:
But in this case, let's leave it as it is. We don't care.